src/test/shared/utilities/functionUtils.test.ts

```diff
@@ -18,6 +18,36 @@ describe('once', function () {
         fn()
         assert.strictEqual(counter, 1)
     })
+
+    it('onceChanged()', function () {
+        let counter = 0
+        const arg2 = {}
+        const arg2_ = { a: 42 }
+        const fn = onceChanged((s: string, o: object) => void counter++)
+
+        fn('arg1', arg2)
+        assert.strictEqual(counter, 1)
+
+        fn('arg1', arg2)
+        fn('arg1', arg2)
+        assert.strictEqual(counter, 1)
+
+        fn('arg1_', arg2)
+        assert.strictEqual(counter, 2)
+
+        fn('arg1_', arg2)
+        fn('arg1_', arg2)
+        fn('arg1_', arg2)
+        assert.strictEqual(counter, 2)
+
+        fn('arg1', arg2_)
+        assert.strictEqual(counter, 3)
+
+        // TODO: bug/limitation: Objects are not discriminated.
+        // TODO: use lib?: https://github.com/anywhichway/nano-memoize
```

Comment:
`nano-memoize` looks reasonable. I didn't bother w/ handling object keys because there are several interpretations for object equality whereas primitives are always compared by value.

I think it makes sense though to start treating object literals more like primitives. This seems to be the trend.